code/frameworks/react-vite/src/plugins/docgen-handlers/actualNameHandler.ts

```diff
@@ -11,32 +11,27 @@
 
 import { namedTypes as t } from 'ast-types';
 import type { NodePath } from 'ast-types/lib/node-path';
-import { getNameOrValue, isReactForwardRefCall } from 'react-docgen/dist/utils';
-import type { Importer } from 'react-docgen/dist/parse';
-import type Documentation from 'react-docgen/dist/Documentation';
+import { utils } from 'react-docgen';
+import type { Documentation } from 'react-docgen';
 
-export default function actualNameHandler(
-  documentation: Documentation,
-  path: NodePath,
-  importer: Importer
-): void {
+export default function actualNameHandler(documentation: Documentation, path: NodePath): void {
   if (t.ClassDeclaration.check(path.node) || t.FunctionDeclaration.check(path.node)) {
-    documentation.set('actualName', getNameOrValue(path.get('id')));
+    documentation.set('actualName', utils.getNameOrValue(path.get('id')));
   } else if (
     t.ArrowFunctionExpression.check(path.node) ||
     t.FunctionExpression.check(path.node) ||
-    isReactForwardRefCall(path, importer)
+    utils.isReactForwardRefCall(path)
   ) {
     let currentPath = path;
     while (currentPath.parent) {
       if (t.VariableDeclarator.check(currentPath.parent.node)) {
-        documentation.set('actualName', getNameOrValue(currentPath.parent.get('id')));
+        documentation.set('actualName', utils.getNameOrValue(currentPath.parent.get('id')));
```

Comment:
These `.set` calls are not allowed by TypeScript. This does not seem like the right thing to do to me.

Reply:
This file is largely copied from https://github.com/reactjs/react-docgen/blob/v6.0.0-alpha.3/src/handlers/displayNameHandler.ts, with a few adjustments.